tracee-rules/engine/engine.go

```diff
@@ -107,6 +106,7 @@ func (engine *Engine) consumeSources(done <-chan bool) {
 							break
 						}
 					}
+					close(sigCh)
```

Comment:
what if there are multiple input channels?

Reply:
Is that possible today? 
https://github.com/aquasecurity/tracee/blob/5cc1e8cc27d03b60c7178995a3e448337815630a/tracee-rules/input.go#L33-L44 

Returns at most one input channel back that gets assigned to be used.

In the case of having the ability to support multiple input channels, I think we need to rethink the design of consumeSources as a whole. Maybe have a goroutine for each input source processing events. But until then I believe this should suffice.

Reply:
If I understand correctly, the panic is caused due to a race condition when a SIGINT or SIGTERM are recieved.
In #777, ^C was hit, which sends SIGINT.
When this happens, the done channel receives 'true' (see sigHandler() in main.go), and at this point two things happen in parallel:
1. The `Start()` function ends, which triggers all the defer close(c)
2. `consumeSources() `also receives 'done'.
The problem is that `consumeSources()` might be in the middle of handling other event, and try to send the event  through one of the sigs channels.

So, my suggestion is to close the signature channels in the `case <-done:` below.
This way, there is no problem if there will be multiple input channels.

Reply:
Hey @yanivagman I think your comment is spot on. I actually bumped into similar error while running benchmarks, not SIGINT or SIGTERM interruption.

However, I do agree that the problem is not with deferred syntax to close the channels created for each signature, but with two gorutines competing over these channels:
1. One that invoked the Start method and deferred close methods. (I assume Start and deferred close methods run in the same thread.)
2. consumeSources goroutine that may send event to channel already closed by 1 in parallel.

Reply:
https://golang.org/doc/articles/race_detector#Unsynchronized_send_and_close_operations